main.go

```diff
@@ -352,12 +353,33 @@ func getManager(
 		os.Exit(1)
 	}
 
-	if err := (&templatesync.PolicyReconciler{
-		Client:   mgr.GetClient(),
-		Scheme:   mgr.GetScheme(),
-		Config:   mgr.GetConfig(),
-		Recorder: mgr.GetEventRecorderFor(templatesync.ControllerName),
-	}).SetupWithManager(mgr); err != nil {
+	depReconciler, depEvents := depclient.NewControllerRuntimeSource()
+
+	watcher, err := depclient.New(managedCfg, depReconciler, nil)
+	if err != nil {
+		log.Error(err, "Unable to create dependency watcher")
+		os.Exit(1)
+	}
+
+	templateReconciler := &templatesync.PolicyReconciler{
+		Client:         mgr.GetClient(),
+		DynamicWatcher: watcher,
+		Scheme:         mgr.GetScheme(),
+		Config:         mgr.GetConfig(),
+		Recorder:       mgr.GetEventRecorderFor(templatesync.ControllerName),
+	}
+
+	go func() {
+		err := watcher.Start(mgrCtx)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	// Wait until the dynamic watcher has started.
+	<-watcher.Started()
```

Comment:
You could move this to right before `return mgr` so that the rest of the setup can continue while dynamic watcher is starting.